libraries/node-core-library/src/FileSystem.ts

```diff
@@ -521,4 +774,34 @@ export class FileSystem {
   public static getRealPath(linkPath: string): string {
     return fsx.realpathSync(linkPath);
   }
+
+  /**
+   * An async version of {@link FileSystem.getRealPath}.
+   */
+  public static getRealPathAsync(linkPath: string): Promise<string> {
+    return LegacyAdapters.convertCallbackToPromise(fsx.realpath, linkPath);
+  }
+
+  // ===============
+  // UTILITY FUNCTIONS
+  // ===============
+
+  /**
+   * Returns true if the error provided indicates the file or folder
+   * does not exist.
+   *
+   * @internal
+   */
+  public static _isNotExistError(error: NodeJS.ErrnoException): boolean {
+    return error.code === 'ENOENT';
```

Comment:
We should also include `error.code === 'ENOTDIR'` in this bucket.

In practice I think we are better off rethrowing all errors that aren't `ENOENT`, `ENOTDIR` or `EEXIST`